cli/progress/multiprogress.go

```diff
@@ -222,63 +328,101 @@ func (m modelMultiProgress) Update(msg tea.Msg) (tea.Model, tea.Cmd) {
 	}
 }
 
-func (m modelMultiProgress) updateOverallProgress() tea.Cmd {
+func (m *modelMultiProgress) allDone() bool {
+	finished := 0
+	for k := range m.Progress {
+		if k == overallProgressIndexName {
+			continue
+		}
+		m.lock.Lock()
+		if m.Progress[k].Errored || m.Progress[k].Completed {
+			finished++
+		}
+		m.lock.Unlock()
+	}
+	allDone := false
+	if _, ok := m.Progress[overallProgressIndexName]; ok {
+		if finished == len(m.Progress)-1 {
+			m.lock.Lock()
+			m.Progress[overallProgressIndexName].Completed = true
+			m.lock.Unlock()
+		}
+		allDone = m.Progress[overallProgressIndexName].Completed
+	} else {
+		allDone = finished == len(m.Progress)
+	}
+
+	return allDone
+}
+
+func (m *modelMultiProgress) updateOverallProgress() {
 	if _, ok := m.Progress[overallProgressIndexName]; !ok {
-		return nil
+		return
 	}
 	overallPercent := 0.0
-	m.Progress[overallProgressIndexName].lock.Lock()
+	m.lock.Lock()
```

Comment:
this method could probably be simplified by locking and the top and defer unlocking right after it